### PR TITLE
i18n utility script

### PR DESCRIPTION
**Pull request by nr-kkenney**
## Summary

This PR includes the i18n utility script to:
* list orphaned translation related files
* delete orphaned files
* list rename changes for locales that need to be made
* make rename changes

Additionally, it includes other changes needed to get testing setup for typescript -- changing the jest config, adding more packages to package.json, updating snapshots.

Example usage:
``` sh
// list orphans
> yarn i18n-utility list-orphans

// delete orphans
> yarn i18n-utility delete-orphans

// list renames
> yarn i18n-utility list-renames

// make renames
> yarn i18n-utility make-renames
```

## Links
Resolves: #5465

### CODE REVIEW

File: scripts/i18n_utility/actions.ts
Comment (roadlittledawn) [inline scripts/i18n_utility/actions.ts:157]: I suppose we could create a ticket to store all relevant language and locale codes in one file, so it's a little easier adding a new language. I shall create that ticket and we can slot it in during adding ko-KR

**Comment by roadlittledawn** [comment]
Just pinged / requested a tech docs person to also take a look at the PR, take it for a spin, read the README

File: scripts/i18n_utility/README.md
Comment (paperclypse) [inline scripts/i18n_utility/README.md:3]: I'd change "CLI" to "command-line" here

I'd also change "for" to "to" so that you can so you can use the infinitive (for example, to: identify, to: delete, to: identify, etc)

File: scripts/i18n_utility/README.md
Comment (paperclypse) [inline scripts/i18n_utility/README.md:2]: I think we could use an intro paragraph here that provides some larger context. It could cover:

1. How the translation files are created and map to the English files/directory structure
2. What happens when there's a mismatch between the translated directories and the English site content
3. What "I18n" means (I assume this is a funny shorthand for "internationalization" which, ick :D )
4. I'm sure there's something else I'm not considering

File: scripts/i18n_utility/README.md
Comment (paperclypse) [inline scripts/i18n_utility/README.md:9]:
```suggestion
Orphaned files are files in `i18n/content/docs` that don't have an English equivalent. They're unused and safe to delete.
```

File: scripts/i18n_utility/README.md
Comment (paperclypse) [inline scripts/i18n_utility/README.md:11]: I was briefly baffled by this paragraph. I'm sure we can clarify and simplify this explanation. Also, why are we using .txt for examples when we work entirely in .mdx files :)

"When you rename a doc's filename, you need to make sure the translated file is renamed to match.

/docs/content/../hello-world.mdx   --> /docs/content/../goodbye-world.mdx  

/docs/content/../i18n/../hello-world.mdx --> /docs/content/../i18n/../goodbye-world.mdx 
"

Or whatever the paths are

File: scripts/i18n_utility/README.md
Comment (paperclypse) [inline scripts/i18n_utility/README.md:13]: I'd change this header to "Troubleshooting"

File: scripts/i18n_utility/README.md
Comment (paperclypse) [inline scripts/i18n_utility/README.md:17]: I'd change this to describe the the situation "You see a rename change you didn't make"

File: scripts/i18n_utility/README.md
Comment (paperclypse) [inline scripts/i18n_utility/README.md:23]: "Your rename isn't detected by git"

File: scripts/i18n_utility/README.md
Comment (paperclypse) [inline scripts/i18n_utility/README.md:25]: Personally, it'd be great to have a better understanding of why this happens (changes not being registered by git).

File: scripts/i18n_utility/README.md
Comment (paperclypse) [inline scripts/i18n_utility/README.md:27]: Should we really be using `git mv` when renaming files/folders?

Comment (nr-kkenney) [reply scripts/i18n_utility/README.md:27]: it depends. if you dont use git mv, git will naturally see changes as add & delete, until you stage them. at that point, git may see them as rename changes. if you make changes & move a file, git may not be able to detect that you renamed a file.

when you use `git mv` you explicitly tell git that you renamed a file, and the change is staged.

File: scripts/i18n_utility/README.md
Comment (nr-kkenney) [reply scripts/i18n_utility/README.md:25]: if the explanation below is good to you, i can add that to the readme.

File: scripts/i18n_utility/README.md
Comment (nr-kkenney) [reply scripts/i18n_utility/README.md:11]: just to nit pick, this change isn't limited to mdx files, but everything else i agree with 🙈 

any file that lives under the i18n content directory is a candidate. .txt is probably an inferior example compared to .mdx, but a fair example would be images.

**Comment by nr-kkenney** [comment]
@paperclypse made the changes you suggested. ready for another round of 👀

File: scripts/i18n_utility/actions.ts
Comment (nr-kkenney) [reply scripts/i18n_utility/actions.ts:157]: i 100% agree with this. outside of this script, this is good to have since there are like 6+ places that use hardcoded versions of locale that we should really have specified in only one place.

Comment (roadlittledawn) [reply scripts/i18n_utility/actions.ts:157]: cool cool. created the ticket. @nr-kkenney feel free to edit / add to it as needed. https://github.com/newrelic/docs-website/issues/5877

**Review by aswanson-nr — APPROVED** [review]

**Review by roadlittledawn — APPROVED** [review]
Took for a test drive myself. Good stuff!

**Comment by aswanson-nr** [comment]
Just waiting on an ✅ from @paperclypse before we merge

**Review by paperclypse — APPROVED** [review]

**Pull request merged**
